### PR TITLE
[MNT-23194] add long validators, correct filename typos

### PR DESCRIPTION
**Pull request by g-jaskowski**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [x] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
https://hyland.atlassian.net/browse/MNT-23194
Types d:int and d:long types have the exact same validators, causing d:long to have identical restrictions as d:int.

**What is the new behaviour?**
Added d:long type validators so the properties of this type accept wider range of number than Integer range.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/content-metadata/services/property-groups-translator.service.spec.ts [hunk 1]
```diff
@@ -428,5 +429,27 @@ describe('PropertyGroupTranslatorService', () => {
...
         });
+
+        it('should translate properly the value attribute for d:long', () => {
+            property.dataType = 'd:long';
+
+            propertyValues = { 'FAS:PLAGUE': '1024' };
+            const cardViewGroup = service.translateToCardViewGroups(propertyGroups, propertyValues, null);
+
+            const cardViewProperty: CardViewLongItemModel = cardViewGroup[0].properties[0] as CardViewLongItemModel;
+            expect(cardViewProperty instanceof CardViewLongItemModel).toBeTruthy();
```
Comment (swapnil-verma-gl) [inline lib/content-services/src/lib/content-metadata/services/property-groups-translator.service.spec.ts:440]: I'm not so sure if this assertion is really needed. The cardViewProperty variable is explicitly defined as a type of CardViewLongItemModel, and then the value being assigned to it during initialising is also type casted as CardViewLongItemModel. So having an assertion to verify its type is not validating the code in the component, rather the code in the test case

File: lib/core/src/lib/card-view/models/card-view-longitem.model.spec.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        expect(itemModel.isValid('21')).toBe(true, 'For "21" it should be true');
+        expect(itemModel.isValid('21.0')).toBe(true, 'For "21.0" it should be true');
+        expect(itemModel.isValid('2e1')).toBe(true, 'For "2e1" it should be true');
+        expect(itemModel.isValid('2g1')).toBe(false, 'For "2g1" it should be false');
+        expect(itemModel.isValid(21.3)).toBe(false, 'For 21.3 it should be false');
+        expect(itemModel.isValid('21.3')).toBe(false, 'For "21.3" it should be false');
+        expect(itemModel.isValid('text')).toBe(false, 'For "text" it should be false');
+    });
+
+    it('should validate based on defined constraints', () => {
```
Comment (swapnil-verma-gl) [inline lib/core/src/lib/card-view/models/card-view-longitem.model.spec.ts:59]: Maybe another test case/assertion that should validate a value between the minimum and maximum, and return true

File: lib/core/src/lib/card-view/validators/card-view-item-long.validator.ts [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+export class CardViewItemLongValidator implements CardViewItemValidator {
+    message = 'CORE.CARDVIEW.VALIDATORS.LONG_VALIDATION_ERROR';
+
+    isValid(value: any | any[]): boolean {
+        if (Array.isArray(value)) {
+            return value.every(this.isLongNumber);
+        }
+
+        return value === '' || (!isNaN(value) && this.isLongNumber(value) && this.isNotEmpty(value));
```
Comment (swapnil-verma-gl) [inline lib/core/src/lib/card-view/validators/card-view-item-long.validator.ts:28]: Do we really need the isNotEmpty() check? If the passed value was a blank string, wouldn't the method return true, on the basis of the initial value === '' check

File: lib/content-services/src/lib/content-metadata/services/property-groups-translator.service.spec.ts (hunk 1 above)
Comment (g-jaskowski) [reply lib/content-services/src/lib/content-metadata/services/property-groups-translator.service.spec.ts:439]: @swapnil-verma-gl You're right, checking instanceof is pointless with type defined. My bad, it was copy-pasted. Fixed similar occurrences in the whole file.

File: lib/core/src/lib/card-view/models/card-view-longitem.model.spec.ts (hunk 2 above)
Comment (g-jaskowski) [reply lib/core/src/lib/card-view/models/card-view-longitem.model.spec.ts:59]: added assertion with correct length

File: lib/core/src/lib/card-view/models/card-view-longitem.model.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+ */
+
+import { CardViewItem } from '../interfaces/card-view-item.interface';
+import { DynamicComponentModel } from '../../common/services/dynamic-component-mapper.service';
+import { CardViewTextItemModel } from './card-view-textitem.model';
+import { CardViewIntItemProperties } from '../interfaces/card-view.interfaces';
+import { CardViewItemLongValidator, CardViewItemPositiveLongValidator } from '../validators/card-view.validators';
+
+export class CardViewLongItemModel extends CardViewTextItemModel implements CardViewItem, DynamicComponentModel {
+    type: string = 'long';
```
[the remark this answers is not in the document]
Comment (g-jaskowski) [reply lib/core/src/lib/card-view/models/card-view-longitem.model.ts]: removed in this and alleother card-view models

File: lib/core/src/lib/card-view/validators/card-view-item-only-positive-long.validator.ts [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+    isValid(value: any | any[]): boolean {
+        if (Array.isArray(value)) {
+            return value.every(this.isPositiveNumber);
+        }
+
+        return value === '' || (!isNaN(value) && this.isPositiveNumber(value));
+    }
+
+    private isPositiveNumber(value: any): boolean {
```
[the remark this answers is not in the document]
Comment (g-jaskowski) [reply lib/core/src/lib/card-view/validators/card-view-item-only-positive-long.validator.ts]: seems to be working correctly as number, changed

**Comment by g-jaskowski** [comment]
Added method allowing different notations of integer number like 1e2 or 1.0 accepted by validators to be used in query. Previously accepted value, but threw a query error. Values like 1.1 are not accepted for d:int and d:long at validators level.

**Review by swapnil-verma-gl — APPROVED** [review]
